src/openloco/companymgr.cpp

```diff
@@ -61,13 +66,46 @@ namespace openloco::companymgr
             if (_byte_525FCB >= 192)
             {
                 _byte_525FCB = 0;
-                sub_4306D1();
+                produce_companies();
             }
         }
     }
 
-    static void sub_4306D1()
+    static void sub_42F9AC()
+    {
+        call(0x0042F9AC);
+    }
+
+    // 0x004306D1
+    static void produce_companies()
     {
-        call(0x004306D1);
+        if (!_company_competition_delay && _company_max_competing)
```

Comment:
`(_company_competition_delay == 0) && (_company_max_competing != 0)` These are not bools so it would be best to stick to comparing them to 0 rather than making the implicit conversion to bool.